core/services/ocr2/delegate.go

```diff
@@ -75,12 +84,12 @@ func NewDelegate(
 	monitoringEndpointGen telemetry.MonitoringEndpointGenerator,
 	chainSet evm.ChainSet,
 	lggr logger.Logger,
-	cfg validate.Config,
+	cfg Config,
 	ks keystore.OCR2,
 	dkgSignKs keystore.DKGSign,
 	dkgEncryptKs keystore.DKGEncrypt,
 	ethKs keystore.Eth,
-	relayers map[relay.Network]types.Relayer,
+	relayers map[relay.Network]func() (loop.Relayer, error),
```

Comment:
this signature is changing, but i don't see any changed calls to NewDelegate. where is this func called? confused way the code is complying...

Reply:
From `application.go`. The line that makes the call didn't need to change.

Reply:
i must be missing something. how is it possible change the type in the map without changing changing the caller to use that new type?